src/view/TrackView.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+
+from typing import List, Optional, Type, Dict, Any
+
+import pygame
+from planar import Vec2
+from pygame.event import EventType
+from pygame.rect import Rect
+from pygame.surface import Surface
+
+from model.Simulation import Simulation
+from model.geom.track import Track
+from view import Colors
+from view.Action import Action, ActionType
+from view.View import View
+
+
+class TrackView(View):
+    simulation: Simulation
+    simulation_class: Type[Simulation]
+    dataset: Dict[str, Any]
+    board: Surface
+    coord_start: Vec2
+    scale = 1.0
+    poi = Vec2(0, 0)
+    # margin = 0
+    background_color = Colors.GRAY
+    foreground_color = Colors.BLACK
+    car_color = Colors.RED
+
+    def __init__(self, simulation: Optional[Simulation]):
+        super().__init__()
+        # to make non optional, as it really is, builder would be required
+        self.simulation = simulation
+
+    @classmethod
+    def from_dataset(
+        cls, simulation_class: Type[Simulation], dataset: Dict[str, Any]
+    ) -> TrackView:
+        tv = cls(None)
+        tv.simulation_class = simulation_class
+        tv.dataset = dataset
+        return tv
+
+    def draw(self, destination: Surface, events: List[EventType]) -> Optional[Action]:
+        if x := self._process_events(events):
+            return x
+        # TODO draw cars/optimisation
+        board = self.board.copy()
+        for car in self.simulation.get_positions():
+            pygame.draw.circle(
+                board, self.car_color, (car.position_x, car.position_y), 10
+            )
+
+        # may be optimised
+        if (
+            destination.get_width() > board.get_width()
+            or destination.get_height() > board.get_height()
+        ):
+            board = self.board
+            new_size = (
+                max(board.get_width(), destination.get_width()),
+                max(board.get_height(), destination.get_height()),
+            )
+            anchor = Vec2(*new_size)
+            anchor -= board.get_size()
+            anchor /= 2
+            new_board = Surface(new_size)
+            new_board.fill(self.background_color)
+            new_board.blit(board, Rect(anchor, self.board.get_size()))
+            board = new_board
+
+        view_rect: Rect = destination.get_rect()
+        limit_x = destination.get_width() // 2
+        limit_y = destination.get_height() // 2
+        center_x = min(max(limit_x, self.poi.x), board.get_width() - limit_x)
+        center_y = min(max(limit_y, self.poi.y), board.get_height() - limit_y)
+        view_rect.center = (int(center_x), int(center_y))
+
+        destination.blit(board.subsurface(view_rect), (0, 0))
+
+        return None
+
+    def activate(self) -> None:
+        super().activate()
+        if not self.simulation:
+            self.simulation = self.simulation_class(**self.dataset)
+        self._prepare_board()
+        pygame.key.set_repeat(350, 50)
```

Comment:
Please add some documentation.

Reply:
`activate` is a function from `View` base class and the documentation should be over there (it is not there yet)